privacy-protections-popup/privacy-protections-popup-impl/src/main/java/com/duckduckgo/privacyprotectionspopup/impl/db/ToggleUsageTimestamp.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.privacyprotectionspopup.impl.db
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import org.threeten.bp.Instant
+
+@Entity(tableName = "toggle_usage_timestamp")
+data class ToggleUsageTimestamp(
+    @PrimaryKey val id: Long = 0,
+    val timestamp: Instant,
+) {
+    init {
+        require(id == 0L)
```

Comment:
This pattern isn't used elsewhere, we use `@PrimaryKey val id: Long = 1` or `@PrimaryKey val id: Int = 1` to indicate that there is only 1 entry.

Ensuring that there is only ever one row can be enforced via unit tests.

Reply:
This check isn't a replacement for unit tests. This class exposes a public constructor that accepts any value of `id` as a parameter, but we don't expect anyone to ever pass a value different than the default one (because that would put our database in an invalid state). Having this contract written down like this just adds a tiny bit of extra safety, but it's a very minor thing, so I can remove it for consistency.

Reply:
I don't feel too strongly about this, it's just different from what we've been doing elsewhere. 
So it's just a consistency thing, I'll let you decide what you want to do here.